program/src/state.rs

```diff
@@ -325,74 +277,62 @@ pub mod tests {
-        assert_eq!(deserialized.authentication, vec![]);
-        assert_eq!(deserialized.capability_invocation, vec![]);
-        assert_eq!(deserialized.capability_delegation, vec![]);
-        assert_eq!(deserialized.key_agreement, vec![]);
-        assert_eq!(deserialized.assertion_method, vec![]);
+        assert_eq!(deserialized.authentication, vec![] as Vec<String>);
+        assert_eq!(deserialized.capability_invocation, vec![] as Vec<String>);
+        assert_eq!(deserialized.capability_delegation, vec![] as Vec<String>);
+        assert_eq!(deserialized.key_agreement, vec![] as Vec<String>);
+        assert_eq!(deserialized.assertion_method, vec![] as Vec<String>);
         assert_eq!(deserialized.service, vec![]);
     }
 
-    #[test]
-    fn parse_did() {
-        let valid_pubkey =
-            Pubkey::from_str("FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP").unwrap();
-
-        let valid = "did:solid:devnet:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP";
-        let did = DecentralizedIdentifier::from_str(&valid).unwrap();
-        assert_eq!(did.cluster_type, ClusterType::Devnet);
-        assert_eq!(did.pubkey, valid_pubkey);
-        assert_eq!(did.identifier, "");
-
-        let valid = "did:solid:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP";
-        let did = DecentralizedIdentifier::from_str(&valid).unwrap();
-        assert_eq!(did.cluster_type, ClusterType::MainnetBeta);
-        assert_eq!(did.pubkey, valid_pubkey);
-        assert_eq!(did.identifier, "");
-
-        let valid = "did:solid:testnet:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP#key1";
-        let did = DecentralizedIdentifier::from_str(&valid).unwrap();
-        assert_eq!(did.cluster_type, ClusterType::Testnet);
-        assert_eq!(did.pubkey, valid_pubkey);
-        assert_eq!(did.identifier, "key1");
-
-        let valid = "did:solid:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP#key1";
-        let did = DecentralizedIdentifier::from_str(&valid).unwrap();
-        assert_eq!(did.cluster_type, ClusterType::MainnetBeta);
-        assert_eq!(did.pubkey, valid_pubkey);
-        assert_eq!(did.identifier, "key1");
-    }
-
-    #[test]
-    fn parse_invalid_did() {
-        // no did:solid
-        let invalid = "solid:devnet:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP";
-        assert_eq!(
-            DecentralizedIdentifier::from_str(&invalid).unwrap_err(),
-            SolidError::InvalidString
-        );
-
-        // unknown network
-        let invalid = "did:solid:mynetwork:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP";
-        assert_eq!(
-            DecentralizedIdentifier::from_str(&invalid).unwrap_err(),
-            SolidError::InvalidString
-        );
-
-        // bad pubkey
-        let invalid = "did:solid:FcFhBFRf6smQ48p7jFcE35uNuE9ScuUu6R2rdFtWjWhP111111";
-        assert_eq!(
-            DecentralizedIdentifier::from_str(&invalid).unwrap_err(),
-            SolidError::InvalidString
-        );
-    }
+    // #[test]
```

Comment:
@joncinque I didn't remove all this yet, but basically I think on-chain parsing and construction of DIDs themselves is no longer necessary or possible (because we don't store the cluster name on chain anymore). I didn't delete it yet just in case I was wrong.